### ISSUE TITLE
False-negative: ParamSpec.args annotating non-*args

### ISSUE THREAD

**Issue by ikonst** [python/mypy#14622]
Per [the docs](https://docs.python.org/3/library/typing.html#typing.ParamSpec),  "P.args ... should only be used to annotate *args":

```python
from typing import ParamSpec

P = ParamSpec('P')


def f(args: P.args) -> None:
    g(*args)


def g(a: str) -> None:
    print(type(a))
    
    

f(42)
```

Gist:
https://mypy-play.net/?mypy=latest&python=3.11&gist=28f03ed1b30e9707edfd080095b6a868

**Comment by erictraut** [comment]
The code above (in particular, the signature for function `f`) isn't legal. A `P.args` annotation can be used only within a context where `P` is bound to a scope (a generic function or class), only with a `*args` parameter, and only when paired with a corresponding `**kwargs: P.kwargs` parameter. See PEP 612 for details.

Mypy should flag the illegal use of `P.args` as an error. (That makes this a false negative, not a false positive.)

**Comment by ikonst** [comment]
@erictraut, thanks for jumping in so quickly!

I've read this in [the docs](https://docs.python.org/3/library/typing.html#typing.ParamSpec)
> P.args represents the tuple of positional parameters in a given call

and failed to read the rest of it :)
> [...] and should only be used to annotate *args

Indeed, then it shall be a false negative.
